src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException; // Правильний імпорт винятку
+import core.basesyntax.model.User;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationService registrationService;
+    private StorageDaoImpl storage;
+
+    @Before
+    public void setUp() {
+        storage = new StorageDaoImpl(); // Використовуємо справжню реалізацію
+        registrationService = new RegistrationServiceImpl(storage);
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User("validLogin", "validPassword", 20);
+
+        registrationService.register(validUser);
+
+        assertEquals(validUser, storage.get(validUser.getLogin()));
+    }
+
+    @Test(expected = RegistrationException.class)
+    public void register_nullUser_notOk() {
+        registrationService.register(null);
+    }
```

Comment:
Instead of using the `expected` parameter in the `@Test` annotation, consider using `assertThrows` to check for exceptions. This way, you can also verify the exception message, which is important for informative error reporting.